HTTPlayground/base.py

```diff
@@ -1,42 +1,59 @@
+from functools import wraps
 from http.cookies import SimpleCookie
 from pathlib import Path
-from HTTPlayground.settings import BASE_DIR, TEMPLATE_DIR, STATIC_URL
+from mimetypes import guess_type
+from HTTPlayground.settings import BASE_DIR, STATIC_URL, ACCESSORY_URL_PREFIX
+
+
+def require_http_methods(request_method_list, error_handler):
+    def decorator(func):
+        @wraps(func)
+        def inner(request, *args, **kwargs):
+            if request.command not in request_method_list:
+                return error_handler(request, *args, **kwargs)
+            return func(request, *args, **kwargs)
+        return inner
+    return decorator
 
 
 def send_headers(request, response_code: int = 200, content_type: str = 'text/plain', **kwargs):
     request.send_response(response_code)
     request.send_header('Content-type', content_type)
+    request.send_header('Access-Control-Allow-Credentials', 'true')
```

Comment:
Looks like:
1. you use this globally for all endpoints of a server
1. you set headers for the other server, even if the current server is the charge/auth server.

I'd be picky about setting cross origin headers because they have a lot to do with safety. In fact you only need those headers in a single endpoint of the second (charge) server.